dimscord/helpers/message.nim

```diff
@@ -0,0 +1,113 @@
+        allowed_mentions = none AllowedMentions;
+        tag: static[bool] = false;
+        tts: static[bool] = false): Future[Message] =
+    ## Replies to a Message.
+    ## (?) - set `tag` to `true` in order to tag the replied message in Discord.
+ 
+    getClient.api.sendMessage(
+        m.channel_id,
+        content, tts, none(int),
+        files, embeds, attachments,
+        allowed_mentions, 
+        when tag == true: some(MessageReference(message_id: some m.id, failIfNotExists: some false)) else: none(MessageReference), 
+        components, stickers
+    )
+
+template edit*(m: Message;
+        content = "";
+        embeds: seq[Embed] = @[];
+        attachments: seq[Attachment] = @[];
+        components: seq[MessageComponent] = @[];
+        files: seq[DiscordFile] = @[];
+        tts: bool = false;
+        flags = none int
+        ): Future[Message]  =
+    ## Edits a Message.
+    ## - Using `edit` for ephemeral messages results in an "Unknown Message" error !
+
+    getClient.api.editMessage(
+        m.channel_id, m.id,
+        content, tts, flags,
+        files, embeds, attachments,
+        components
+    )
+
+template delete*(m: Message, reason = ""): Future[void] =
+    ## Deletes a Message.
+    getClient.api.deleteMessage(m.channel_id, m.id, reason)
+
+template react*(m: Message, emoji: string): Future[void] =
+    ## Add a reaction to a Message
+    ##
+    ## - `emoji` Example: '👀', '💩', `likethis:123456789012345678`
+    getClient.api.addMessageReaction(m.channel_id, m.id, emoji)
+
+template unreact*(m: Message, emoji: string, user_id = "@me"): Future[void] =
+    ## Deletes the user's or the bot's message reaction to a Discord message.
+    getClient.api.deleteMessageReaction(m.channel_id, m.id, emoji, user_id)
+
+template unreactAll*(m: Message, emoji: string): Future[void] =
+    ## Deletes all the reactions for emoji.
+    getClient.api.deleteMessageReactionEmoji(m.channel_id, m.id, emoji)
+
+template reactions*(m: Message, emoji: string;
+        before, after = "";
+        limit: range[1..100] = 25
+): Future[seq[User]] =
+    ## Get all user message reactions on the emoji provided.
+    getClient.api.getMessageReactions(
+        m.channel_id, m.id, emoji,
+        before, after, limit
+    )
+template clear*(m: Message): Future[void] =
+    ## Remove all message reactions.
+    getClient.api.deleteAllMessageReactions(m.channel_id, m.id)
+
+template leave*(ch: GuildChannel): Future[void] =
+    ## Leave thread.
+    if ch.kind == ctGuildPublicThread or (ch.kind == ctGuildPrivateThread):
+        getClient.api.leaveThread(ch.id)
+    else:
+        raise newException(CatchableError, "Channel is not a thread !")
+
+template join*(ch: GuildChannel): Future[void] =
+    ## Join thread.
+    if ch.kind == ctGuildPublicThread or (ch.kind == ctGuildPrivateThread):
+        getClient.api.joinThread(ch.id)
+    else:
+        raise newException(CatchableError, "Channel is not a thread !")
```

Comment:
As said earlier, I dont think this part is necessary if the joinThread proc fails then it should print out the error message from discord rest api.